apps/web/src/apis/usernames.ts

```diff
@@ -0,0 +1,62 @@
+import { ContractFunctionParameters, namehash } from 'viem';
```

Comment:
imo an `apis` dir indicates backend logic. It looks like these functions are intended for consumption in a non-browser environment, but aren't themselves "APIs" in the sense that this directory will likely be interpreted in the future. IMO these should live nearer their usage in the backend request handlers.

Reply:
These are used by both FE / BE, e.g: 

FE: `apps/web/src/hooks/useReadBaseEnsTextRecords.ts` 
BE: `apps/web/src/app/(basenames)/name/[username]/page.tsx`

I put them here because they work in any env & I didn't want to keep them in `utils/usernames` because they're not really utilities and that file was already long.